src/EnergyPlus/Psychrometrics.cc

```diff
@@ -366,26 +366,16 @@ namespace Psychrometrics {
         // na
 
         // FUNCTION LOCAL VARIABLE DECLARATIONS:
-        Int64 Tdb_tag;
-        Int64 W_tag;
-        Int64 Pb_tag;
-        Int64 hash;
-        Real64 Tdb_tag_r;
-        Real64 W_tag_r;
-        Real64 Pb_tag_r;
 
 #ifdef EP_psych_stats
         ++state.dataPsychrometrics->NumTimesCalled(iPsyTwbFnTdbWPb_cache);
 #endif
 
-        Tdb_tag = bit_transfer(Tdb, Tdb_tag);
-        W_tag = bit_transfer(W, W_tag);
-        Pb_tag = bit_transfer(Pb, Pb_tag);
+        std::uint64_t Tdb_tag = *reinterpret_cast<std::uint64_t const *>(&Tdb) >> Grid_Shift;
```

Comment:
This is the "essence" of this PR. ```bit_transfer``` instances were replaced by ```reinterpret_cast```, and ```bit_shift``` was replaced with the ```<<``` or ```>>``` bitwise operators. However, we're still getting a warning on Linux for these.

> dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]

I'm not sure how to resolve that currently.

Reply:
Option 0. Ignore the warning, it's just a warning. It's the compiler saying "are you sure you know what you're doing?" and in this case we do. At least I do.
Option 1. Compile with `-fno-strict-aliasing` on Linux.
Option 2. This may not be portable, but we can use `#pragma` to locally disable specific warnings and we might want to do that around these specific cache-indexing uses of `reinterpret_cast`, maybe with an `assert(sizeof(Tdb) == sizeof(Tdb_tag));` added in. See https://www.fluentcpp.com/2019/08/30/how-to-disable-a-warning-in-cpp/.

Reply:
It looks like we've already gone down this road and opted for Option 2: https://github.com/NREL/EnergyPlus/blob/5358d3eb87d1376b3484e1a38ade8274b01f0027/src/EnergyPlus/Psychrometrics.hh#L622-L625

Reply:
Ok then.